docs/parser-cxx.rst

```diff
@@ -31,11 +31,151 @@ features were added. Among them:
+
+Several properties of functions and variables can be extracted
+and placed in a new field called ``properties``.
+The syntax to enable it is:
+
+.. code-block:: C
+
+	$ ctags ... --fields-c++=+{properties} ...
+
+At the time of writing the following properties are reported:
+
+- ``virtual``: a function is marked as virtual
+- ``static``: a function/variable is marked as static
+- ``inline``: a function implementation is marked as inline
+- ``explicit``: a function is marked as explicit
+- ``extern``: a function/variable is marked as extern
+- ``const``: a function is marked as const
+- ``pure``: a virtual function is pure (i.e = 0)
+- ``override``: a function is marked as override
+- ``default``: a function is marked as default
+- ``final``: a function is marked as final
+- ``delete``: a function is marked as delete
+- ``mutable``: a variable is marked as mutable
+- ``volatile``: a function is marked as volatile
+- ``specialization``: a function is a template specialization
+- ``scopespecialization``: template specialization of scope ``a<x>::b()``
+- ``deprecated``: a function is marked as deprecated via ``__attribute__``
+- ``scopedenum``: a scoped enumeration (C++11)
+
+Preprocessor macros
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The new parser supports the definition of real preprocessor macros
+via the ``-D`` option. All types of macros are supported,
+including the ones with parameters and variable arguments.
+Stringification, token pasting and recursive macro expansion are also supported.
+
+Option ``-I`` is now simply a backward-compatible syntax to define a
+macro with no replacement.
+
+The syntax is similar to the corresponding gcc ``-D`` option.
+
+Some examples follow.
+
+.. code-block:: console
+
+	$ ctags ... -D IGNORE_THIS ...
+
+With this commandline the following C/C++ input
+
+.. code-block:: C
+
+	int IGNORE_THIS a;
+
+will be processed as if it was
+
+.. code-block:: C
+
+	int a;
+
+Defining a macro with parameters uses the following syntax:
+
+.. code-block:: console
+
+	$ ctags ... -D "foreach(arg)=for(arg;;)" ...
+
+This example defines `for(arg;;)` as the replacement `foreach(arg)`.
+So the following C/C++ input
+
+.. code-block:: C
+
+	foreach(char * p,pointers)
+	{
+
+	}
+
+is processed in new C/C++ parser as:
+
+.. code-block:: C
+
+	for(char * p;;)
+	{
+
+	}
+
+and the p local variable can be extracted.
+
+The previous commandline includes quotes since the macros generally contain
+characters that are treated specially by the shells. You may need some escaping.
+
+Token pasting is performed by the ## operator, just like in the normal
+C preprocessor.
+
+.. code-block:: console
+
+	$ ctags ... -D "DECLARE_FUNCTION(prefix)=int prefix ## Call();"
+
+So the following code
+
+.. code-block:: C
+
+	DECLARE_FUNCTION(a)
+	DECLARE_FUNCTION(b)
+
+will be processed as
+
+.. code-block:: C
+
+	int aCall();
+	int bCall();
+
+Macros with variable arguments use the gcc __VA_ARGS__ syntax.
```

Comment:
might be worth wrapping `__VA_ARGS__` in ``es to avoid it being treated as underline